tests/integration/test_tests.rs

```diff
@@ -177,6 +177,12 @@ itest!(text {
   output: "test/text.out",
 });
 
+itest!(type_check_with_doc {
```

Comment:
Can you move this to the specs directory? This will fail linting because we have a lint that makes sure no more of these tests are added